app/views/scrapers/_danger.html.haml

```diff
@@ -16,9 +16,14 @@
                   of scraped data will be gone!
                 - else
                   All your data will be gone!
-              = button_to clear_scraper_path(@scraper), {class: "btn btn-warning", data: {confirm: "All the data you’ve scraped will be deleted. Are you sure you want to clear your scraper’s database?"}} do
-                Clear database
-                %i.glyphicon.glyphicon-th-list
+              - if @scraper.has_data?
+                = button_to clear_scraper_path(@scraper), {class: "btn btn-warning", data: {confirm: "All the data you’ve scraped will be deleted. Are you sure you want to clear your scraper’s database?"}} do
```

Comment:
You should be able to remove all the repetition of the button html by making the condition only cover the attribute we care about here. Something like `button_to {disabled: ( "disabled" unless @scraper.has_data? )}`